traits/trait_types.py

```diff
@@ -1450,7 +1450,11 @@ def __init__(
 
 
 class BaseDirectory(BaseStr):
-    """ A trait type whose value is a directory path string.
+    """ A trait type whose value must be a directory path string.
+
+    For Python 3.6 and greater, it also accepts objects implementing
+    the :class: `os.PathLike` interface, converting them to the corresponding
```

Comment:
Normal reST style is to leave out the space between the `:class:` and the contents; I'm not sure whether it matters to Sphinx or not.

```suggestion
    the :class:`os.PathLike` interface, converting them to the corresponding
```